### PR TITLE
webauthn + refactor

### PR DESCRIPTION
**Pull request by tantommy**
- adds webauthn
- refactor - rename Identity to Address

### CODE REVIEW

**Comment by hansl** [comment]
Just want to clarify if this is ready for review, as the title is still "NOT READY FOR MERGE".

**Comment by tantommy** [comment]
> Just want to clarify if this is ready for review, as the title is still "NOT READY FOR MERGE".

This is ready for review.  I'll change the title.

File: src/message/cose.ts [hunk 1]
```diff
@@ -81,15 +81,15 @@ export class CoseMessage {
...
     content: CborMap,
-    privateKey: Key
-  ): Buffer {
+    identity: Identity,
+  ): Promise<Buffer | ArrayBuffer> {
     const p = cbor.encodeCanonical(protectedHeader);
     const payload = cbor.encode(tag(10001, content));
     const message = cbor.encodeCanonical(["Signature1", p, EMPTY, payload]);
-    return Buffer.from(ed25519.sign({ message, privateKey }));
+    return await identity.sign(message)
```
[the remark this answers is not in the document]
Comment (hansl) [reply src/message/cose.ts:92]: That is correct. But for the wallet specifically we don't need to sign every requests, only commands (ie. send).

File: src/identity/webauthn/webauthn-identity.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    },
+
+    pubKeyCredParams: [
+      {
+        /*
+          EdDSA	-8
+          ES256	-7	ECDSA w/ SHA-256
+        */
+        type: "public-key",
+        alg: -8,
```
[the remark this answers is not in the document]
Comment (hansl) [reply src/identity/webauthn/webauthn-identity.ts:98]: I think so, but not entirely sure.

File: src/identity/ed25519/ed25519-key-pair-user-identity.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { pki } from "node-forge"
```
[the remark this answers is not in the document]
Comment (hansl) [reply src/identity/ed25519/ed25519-key-pair-user-identity.ts]: Correct.

**Pull request closed without merging**